tt_metal/hw/ckernels/wormhole_b0/metal/llk_api/llk_sfpu/ckernel_sfpu_rand.h

```diff
@@ -0,0 +1,42 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+#pragma once
+
+#include "ckernel.h"
+#include "ckernel_defs.h"
+
+using namespace sfpi;
+
+namespace ckernel::sfpu {
+
+template <bool APPROXIMATION_MODE>
+inline void rand_init(uint seed) {
+    init_prng_seed(seed);
+}
+
+template <bool APPROXIMATION_MODE>
+inline void rand(float from, float to) {
+#pragma GCC unroll 0
+    for (int d = 0; d < 8; d++) {
+        // Generate random float
+        TTI_SFPMOV(0, 9, p_sfpu::LREG0, 8);
+        // Unset sign bit and Set exponent to 127 to ensure the float is within the range [1, 2).
+        TTI_SFPSETSGN(0, p_sfpu::LREG0, p_sfpu::LREG0, 1);
+        TTI_SFPSETEXP(127, p_sfpu::LREG0, p_sfpu::LREG0, 1);
+        // TTI_SFPADDI(0xbfa7, p_sfpu::LREG0, 0);
+        TTI_SFPSTORE(0, 4, 3, 0);
+
+        // -1 to ensure the float is within the range [0, 1).
+        vFloat rand_floats = dst_reg[0];
```

Comment:
In general it is recommended not to mix SFPI and the TTI_* instructions and to write them using only one form. 
Do you need to save to dst and reload from it ? Is it not possible to have all the values in the registers and do one final write to dst ? That would save you some cycles I think.

Reply:
That's what I want to ask you. I tried arithmetic instruction like: `TTI_SFPADDI(1, p_sfpu::LREG0, 0);` before `TTI_SFPSTORE` to -1 to rand_float but the generated numbers are not changed. That's why I have to update use `vFloat`

This is the first time I deal with tensix instruction. Is there anything I'm doing wrong here with the arithmetic instructions.

Reply:
The thing is the compiler will not do anything with the TTI_* instructions and keep them as they are, but it may try to optimize the vFloat instructions, so there might be register reordering or something, but in your case you are writing to dst and then reading from dst, so there is not much for the compiler to move around or change so should be okay. But the thing is when using TTI instructions you know exactly which registers you are using so you may end up fitting all the data within the registers and not need to read and write from the dst for intermediate values saving some cycles.

Reply:
@BuiChiTrung  SFPADDI requires 2 cycles to complete, so you need to put a NOP after it  otherwise the result will be incorrect if an instruction uses it right after.

Reply:
I tried with NOP(), the value is updated. I tried pass various of value as `imm16_math` of SFPADDI param but it only return some 32x32 weird value of -nan.
```
TTI_SFPADDI(-1, p_sfpu::LREG0, 0);
TTI_SFPNOP;
```
If it's possible, could you suggest the code to convert those statements with vFloat to TTI_*. As there is no document with TTI_, it's really hard converting sfpi to TTI_, which requires a lot of trial and error.

Reply:
You need to put the bit pattern of -1 in bfp16 format. So you can save -1 in bfp16 format, print the unsigned int value for that bit pattern and put that value here.

Reply:
Got that. I've update the ckernel.